apis/v1alpha1/gatewayclass_types.go

```diff
@@ -80,30 +80,38 @@ type GatewayClassSpec struct {
 	ParametersRef *LocalObjectReference `json:"parametersRef,omitempty"`
 }
 
-// GatewayClassConditionType is the type of status conditions. This
-// type should be used with the GatewayClassStatus.Conditions field.
+// GatewayClassConditionType is the type of status conditions for Gateway resources.
+// This type should be used with the GatewayClassStatus.Conditions field.
 type GatewayClassConditionType string
 
+// GatewayClassConditionReason defines the set of reasons that explain why
+// a particular GatewayClass condition type has been raised.
+type GatewayClassConditionReason string
+
 const (
-	// GatewayClassConditionStatusAdmitted indicates that the GatewayClass
-	// has been admitted by the controller set in the `spec.controller`
+	// This condition indicates whether the GatewayClass
+	// has been admitted by the controller requested in the `spec.controller`
```

Comment:
Nit: Indentation feels a bit off here.

Reply:
That would be clear if it was done consistently. Maybe we can make a separate pass to do that everywhere.